staging/src/k8s.io/apiserver/pkg/apis/audit/types.go

```diff
@@ -300,3 +301,61 @@ type UserInfo struct {
 
 // ExtraValue masks the value so protobuf can generate
 type ExtraValue []string
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuditConfiguration represents a cluster level audit configuration
+type AuditConfiguration struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	// if undefined it will default to the statically configured cluster policy if available
+	// if neither exist the backend will fail
+	// +optional
+	Policy *Policy
+
+	// Backend to send events
+	Backend Backend
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuditConfigurationList is a list of AuditConfiguration items.
+type AuditConfigurationList struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ListMeta
+	// List of audit configurations.
+	Items []AuditConfiguration
+}
+
+// Backend holds the configuration for the backend
+type Backend struct {
+	// Webhook holds the webhook backend
+	// +optional
+	Webhook *WebhookBackend
+}
+
+// WebhookBackend holds the configuration of the webhooks
+type WebhookBackend struct {
+	// InitialBackoff is amount of time to wait before retrying the first failed request in seconds
+	// +optional
+	InitialBackoff *int64
+
+	// ThrottleBurst is the maximum number of events sent at the same moment
+	// +optional
+	ThrottleBurst *int64
+
+	// ThrottleEnabled determines whether throttling is enabled
+	// +optional
+	ThrottleEnabled *bool
+
+	// ThrottleQPS maximum number of batches per second
+	// +optional
+	ThrottleQPS *float32
+
+	// ClientConfig holds the connection parameters for the webhook
+	// required
+	ClientConfig apiv1alpha1.WebhookClientConfig
```

Comment:
If this is supposed to be a reference to a webhook, shouldn't it be a name? OTOH if this is where the webhook configuration is stored, why is there a top level Webhook type?

Reply:
This is a nested webhook configuration, the top level webhook type was just to keep parity with the original design, it can be scrapped though